### PR TITLE
Greedy candy

### PR DESCRIPTION
**Pull request by siddharthzs**
### **Describe your change:**
Leetcode 135: Distribute Candy
There are N children standing in a line. Each child is assigned a rating value. You are giving candies to these children subjected to the following requirements:
      Each child must have at least one candy. 
      Children with a higher rating get more candies than their neighbors.
What is the minimum candies you must give?

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by jmothiki** [comment]
awesome

File: greedy_method/distribute_candy.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    Input Format:
+        A list of Numbers representing children ratings.
+    Output Format:
+        Minimum number of candies to be given.
+    children_ratings = [1, 2]
+    >>> 3
+    children_ratings = [1, 5, 2, 1]
+    >>> 7
+    children_ratings = [1, 3, 2, 1]
+    >>> 7
```
Comment (xcodz-dot) [inline greedy_method/distribute_candy.py:17]:
```suggestion

    >>> distribute_candy([1, 2])
    3
    >>> distribute_candy([1, 5, 2, 1])
    7
    >>> distribute_candy([1, 3, 2, 1])
    7
```

This is the correct syntax of doctests, i have made the changes for you. please accept them

**Review by xcodz-dot — CHANGES_REQUESTED** [review]

**Comment by siddharthzs** [comment]
Thank you @xcodz-dot, If you wouldn't have helped me I wouldn't be able to pass the test.

**Comment by xcodz-dot** [comment]
This seems to be good to go

@Kush1101 @dhruvmanila, Check this out

**Comment by dhruvmanila** [comment]
This pull request contains an additional file which seems like a mistake. Please make the necessary changes to only include the file containing the `Greedy candy` algorithm or open a new pull request for it.

**Pull request closed without merging**